x-pack/plugins/security_solution/public/assistant/use_assistant_availability/index.tsx

```diff
@@ -16,11 +18,11 @@ export interface UseAssistantAvailability {
 
 export const useAssistantAvailability = (): UseAssistantAvailability => {
   const isEnterprise = useLicense().isEnterprise();
+  const capabilities = useKibana().services.application.capabilities;
+  const isAssistantEnabled = capabilities[ASSISTANT_FEATURE_ID]?.isAssistantEnabled === true;
+
   return {
     isAssistantEnabled: isEnterprise,
-    // TODO: RBAC check (https://github.com/elastic/security-team/issues/6932)
-    // Leaving as a placeholder for RBAC as the same behavior will be required
-    // When false, the Assistant is hidden and unavailable
-    hasAssistantPrivilege: true,
+    hasAssistantPrivilege: isAssistantEnabled,
```

Comment:
Not sure we need two separate pieces of state for tracking license vs capabilities now since the privilege itself has [`minimumLicense: 'enterprise'`](https://github.com/elastic/kibana/pull/163031/files#diff-56de8b6234daf4e0e69efe680e5a4afc4f88d152243b773d90c3991fa9dabc19R28) declared and we have no current need to make a distinction at the assistant layer either. 

Maybe in the future if only a subset of assistant features have license requirements? Will keep for the time being....

Reply:
Then it looks like this hook could return only `hasAssistantPrivilege` and that would suffice. I think also that the returned property `isAssistantEnabled` here is kind of mis-named; I would call it `isAssistantAvailable`, since it depends on the license being enterprise or not. If the assistant is **enabled** or not depends on the priviliges and capabilities, as the code shows.

Reply:
So yeah, I think we're going to want to keep these separate as I forgot we had a subscription upgrade flow within the assistant itself, so there is a distinction between no-license and disabled. That said, I do have an open question [over here](https://github.com/elastic/kibana/pull/163031/files/41796aa79c876616da1d1e10bca5201ab8398914#r1287427542) with regards to how `minimumLicense:` should function on the privilege -- only for new roles, or does `Security:All` override and make it `true` for existing roles?